fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -2527,11 +2528,11 @@ && isNoneOrCashOrUpfrontAccrualAccountingEnabledOnLoanProduct()) {
     }
 
     private void regenerateRepaymentScheduleWithInterestRecalculationIfNeeded(boolean interestRecalculationEnabledParam,
-            boolean disbursementMissedParam, ScheduleGeneratorDTO scheduleGeneratorDTO) {
+            boolean disbursementMissedParam, ScheduleGeneratorDTO scheduleGeneratorDTO, final boolean downPaymentEnabled) {
 
         LocalDate firstInstallmentDueDate = fetchRepaymentScheduleInstallment(1).getDueDate();
-        if (interestRecalculationEnabledParam
-                && (firstInstallmentDueDate.isBefore(DateUtils.getBusinessLocalDate()) || disbursementMissedParam)) {
+        if ((interestRecalculationEnabledParam
+                && (firstInstallmentDueDate.isBefore(DateUtils.getBusinessLocalDate()) || disbursementMissedParam)) || downPaymentEnabled) {
```

Comment:
otherwise the rescheduling won't happen, if the interest recalcualtion is disabled, but we need to reschedule the loan to create the down payment installment

Reply:
I am not sure i like the idea to mix downpayment logic into `regenerateRepaymentScheduleWithInterestRecalculationIfNeeded` method :/

Reply:
I separated the down payment from interest recalculation